src/apps/jsgeneric/jsgeneric.cpp

```diff
@@ -13,33 +13,106 @@ namespace ccfapp
   using namespace kv;
   using namespace ccf;
 
-  using GenericTable = ccf::Store::Map<nlohmann::json, nlohmann::json>;
+  using LogTable = ccf::Store::Map<int32_t, std::string>;
+
+  static JSValue js_print(
+    JSContext* ctx, JSValueConst this_val, int argc, JSValueConst* argv)
+  {
+    int i;
+    const char* str;
+    auto level = logger::INFO;
+
+    if (logger::config::ok(level))
+    {
+      auto os = logger::LogLine(level, __FILE__, __LINE__);
+      for (i = 0; i < argc; i++)
+      {
+        if (i != 0)
+          os << ' ';
+        str = JS_ToCString(ctx, argv[i]);
+        if (!str)
+          return JS_EXCEPTION;
+        os << str;
+        JS_FreeCString(ctx, str);
+      }
+      os << std::endl;
+      auto _ = logger::Out() == os;
```

Comment:
Do we need to deal with the internals of `logger` here? Or can we do something like:

```
std::stringstream ss;
for (i = 0; i < argc; ++i)
{
  if (i != 0) ss << ' ';
  str = JS_ToCString(ctx, argv[i]);
  if (!str) ...
  ss << str;
  JS_FreeCString(ctx, str);
}
LOG_INFO_FMT(ss.str());
```

Reply:
Thanks, I didn't think of that!

Reply:
Done